[DOCS] Clarify CCR auto-follow requirements for data streams (#72476)
elastic/elasticsearch: docs/reference/ccr/apis/auto-follow/put-auto-follow-pattern.asciidoc

`follow_index_pattern`::
  (Optional, string) The name of follower index. The template `{{leader_index}}`
  can be used to derive the name of the follower index from the name of the

>   leader index. When following a data stream, `{{leader_index}}` should be used
>   as changing the names of a follower data stream's backing indices is not
>   supported.

Reviewer: Minor reword to break this into two sentences and avoid passive voice.
```suggestion
  leader index. When following a data stream, use `{{leader_index}}`; {ccr-init}
  does not support changes to the names of a follower data stream's backing
  indices.
  ```
